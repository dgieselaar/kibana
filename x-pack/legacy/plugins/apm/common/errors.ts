/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export enum ErrorStatus {
  ACTIVE = 'active',
  MUTED = 'muted',
  RESOLVED = 'resolved',
  REOCCURED = 'reoccured',
  NEW = 'new'
}

export const ALL_ERRORS = 'all';
export const ACTIVE_ERRORS = 'active';
export const ARCHIVED_ERRORS = 'archived';

export type ErrorStatusGroup =
  | typeof ALL_ERRORS
  | typeof ACTIVE_ERRORS
  | typeof ARCHIVED_ERRORS;
