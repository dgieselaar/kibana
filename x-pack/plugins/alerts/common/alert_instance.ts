/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import * as t from 'io-ts';
import { DateFromString } from './date_from_string';

const metaSchema = t.intersection([
  t.partial({
    lastScheduledActions: t.intersection([
      t.partial({
        subgroup: t.string,
      }),
      t.type({
        group: t.string,
        date: DateFromString,
      }),
    ]),
  }),
  t.type({
    instance: t.intersection([
      t.type({
        uuid: t.string, // unique identifier for this instance
        id: t.string, // rule_id + name
        name: t.string, // unique identifier within the context of rule
        started_at: DateFromString, // start date for this instance
      }),
      t.partial({
        title: t.string, // human-readable label
      }),
    ]),
  }),
]);
export type AlertInstanceMeta = t.TypeOf<typeof metaSchema>;

const stateSchema = t.record(t.string, t.unknown);
export type AlertInstanceState = t.TypeOf<typeof stateSchema>;

const contextSchema = t.record(t.string, t.unknown);
export type AlertInstanceContext = t.TypeOf<typeof contextSchema>;

export const rawAlertInstance = t.intersection([
  t.partial({
    state: stateSchema,
  }),
  t.type({
    meta: metaSchema,
  }),
]);

export type RawAlertInstance = t.TypeOf<typeof rawAlertInstance>;
