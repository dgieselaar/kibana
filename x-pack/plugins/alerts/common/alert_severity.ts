/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

export enum AlertSeverity {
  Low = 'low',
  Warning = 'warning',
  Minor = 'minor',
  Major = 'major',
  Critical = 'critical',
}
