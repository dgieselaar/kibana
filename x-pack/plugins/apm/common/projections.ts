/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export enum PROJECTION {
  SERVICES = 'services',
  TRANSACTION_GROUPS = 'transactionGroups',
  TRACES = 'traces',
  TRANSACTIONS = 'transactions',
  METRICS = 'metrics',
  ERROR_GROUPS = 'errorGroups',
  SERVICE_NODES = 'serviceNodes',
  RUM_OVERVIEW = 'rumOverview',
}
