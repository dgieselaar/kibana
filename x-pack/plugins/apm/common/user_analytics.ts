/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';
import * as t from 'io-ts';

export const userAnalyticsConfig = {
  index: 'google-analytics-data',
  metrics: {
    users: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.users', {
        defaultMessage: 'Users',
      }),
      field: 'user.id',
      type: 'cardinality',
    },
    hits: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.hits', {
        defaultMessage: 'Hits',
      }),
      field: 'session.totals.hits',
      type: 'sum',
    },
    pageLoadTime: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.pageLoadTime', {
        defaultMessage: 'Page load time',
      }),
      field: 'latencyTracking.pageLoadTime',
      type: 'avg',
    },
    pageViews: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.pageViews', {
        defaultMessage: 'Page views',
      }),
      field: 'session.totals.pageviews',
      type: 'sum',
    },
    sessions: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.sessions', {
        defaultMessage: 'Sessions',
      }),
      field: 'session.id',
      type: 'cardinality',
    },
    sessions_per_user: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.sessionsPerUser', {
        defaultMessage: 'Sessions per user',
      }),
      field: 'session.totals.visits',
      type: 'avg',
    },
    bounces: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.bounces', {
        defaultMessage: 'Bounces',
      }),
      field: 'session.totals.bounces',
      type: 'sum',
    },
    transactions: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.transactions', {
        defaultMessage: 'Transactions',
      }),
      field: 'session.totals.transactions',
      type: 'sum',
    },
    revenue: {
      title: i18n.translate('xpack.apm.userAnalytics.metrics.revenue', {
        defaultMessage: 'Revenue',
      }),
      field: 'session.totals.transactionRevenue',
      type: 'sum',
    },
  },
} as const;

export const segmentMetricRt = t.keyof(userAnalyticsConfig.metrics);

export const segmentRt = t.intersection([
  t.type({
    id: t.string,
    title: t.string,
    color: t.string,
    metric: segmentMetricRt,
  }),
  t.partial({
    eql: t.string,
  }),
]);

export type SegmentMetricType = t.TypeOf<typeof segmentMetricRt>;
export type Segment = t.TypeOf<typeof segmentRt>;
