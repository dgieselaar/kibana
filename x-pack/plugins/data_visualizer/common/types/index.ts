/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import { SimpleSavedObject } from '../../../../../src/core/public/saved_objects/simple_saved_object';

export type {
  DocumentCountBuckets,
  DocumentCounts,
  FieldRequestConfig,
  FieldVisStats,
  Percentile,
} from './field_request_config';
export type { JobFieldType } from './job_field_type';
export type InputData = any[];

export interface DataVisualizerTableState {
  pageSize: number;
  pageIndex: number;
  sortField: string;
  sortDirection: string;
  visibleFieldTypes: string[];
  visibleFieldNames: string[];
  showDistributions: boolean;
}

export type SavedSearchSavedObject = SimpleSavedObject<any>;
