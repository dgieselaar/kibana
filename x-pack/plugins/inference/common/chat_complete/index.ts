/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import type { Observable } from 'rxjs';
import type { InferenceTaskEventBase } from '../tasks';
import type { ToolCall, ToolCallsOf, ToolOptions } from './tools';

export enum MessageRole {
  User = 'user',
  Assistant = 'assistant',
  Tool = 'tool',
}

interface MessageBase<TRole extends MessageRole> {
  role: TRole;
}

export type UserMessage = MessageBase<MessageRole.User> & { content: string };
export type AssistantMessage = MessageBase<MessageRole.Assistant> & {
  content: string | null;
  toolCalls?: Array<ToolCall<string, Record<string, any> | undefined>>;
};

export type ToolMessage<TToolResponse extends Record<string, any> | unknown> =
  MessageBase<MessageRole.Tool> & {
    toolCallId: string;
    response: TToolResponse;
  };

export type Message = UserMessage | AssistantMessage | ToolMessage<unknown>;

export type ChatCompletionMessageEvent<TToolOptions extends ToolOptions> =
  InferenceTaskEventBase<ChatCompletionEventType.ChatCompletionMessage> & {
    content: string;
  } & { toolCalls: ToolCallsOf<TToolOptions>['toolCalls'] };

export type ChatCompletionResponse<TToolOptions extends ToolOptions = ToolOptions> = Observable<
  ChatCompletionEvent<TToolOptions>
>;

export enum ChatCompletionEventType {
  ChatCompletionChunk = 'chatCompletionChunk',
  ChatCompletionTokenCount = 'chatCompletionTokenCount',
  ChatCompletionMessage = 'chatCompletionMessage',
}

export interface ChatCompletionChunkToolCall {
  index: number;
  toolCallId: string;
  function: {
    name: string;
    arguments: string;
  };
}

export type ChatCompletionChunkEvent =
  InferenceTaskEventBase<ChatCompletionEventType.ChatCompletionChunk> & {
    content: string;
    tool_calls: ChatCompletionChunkToolCall[];
  };

export type ChatCompletionTokenCountEvent =
  InferenceTaskEventBase<ChatCompletionEventType.ChatCompletionTokenCount> & {
    tokens: {
      prompt: number;
      completion: number;
      total: number;
    };
  };

export type ChatCompletionEvent<TToolOptions extends ToolOptions = ToolOptions> =
  | ChatCompletionChunkEvent
  | ChatCompletionTokenCountEvent
  | ChatCompletionMessageEvent<TToolOptions>;

export type ChatCompleteAPI<TToolOptions extends ToolOptions = ToolOptions> = (
  options: {
    connectorId: string;
    system?: string;
    messages: Message[];
  } & TToolOptions
) => ChatCompletionResponse<TToolOptions>;
