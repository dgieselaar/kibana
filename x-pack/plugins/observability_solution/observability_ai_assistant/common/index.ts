/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

export type { Message, Conversation, KnowledgeBaseEntry } from './types';
export { KnowledgeBaseEntryRole, MessageRole } from './types';
export { FunctionVisibility } from './functions/types';
export {
  VISUALIZE_ESQL_USER_INTENTIONS,
  VisualizeESQLUserIntention,
} from './functions/visualize_esql';
