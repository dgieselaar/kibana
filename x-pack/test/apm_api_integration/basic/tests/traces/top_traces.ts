/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
import expect from '@kbn/expect';
import { sortBy } from 'lodash';
import archives_metadata from '../../../common/archives_metadata';
import { expectSnapshot } from '../../../common/match_snapshot';
import { FtrProviderContext } from '../../../../common/ftr_provider_context';

export default function ApiTest({ getService }: FtrProviderContext) {
  const supertest = getService('supertest');
  const esArchiver = getService('esArchiver');

  const archiveName = 'apm_8.0.0';
  const metadata = archives_metadata[archiveName];

  // url parameters
  const start = encodeURIComponent(metadata.start);
  const end = encodeURIComponent(metadata.end);
  const uiFilters = encodeURIComponent(JSON.stringify({}));

  describe('Top traces', () => {
    describe('when data is not loaded ', () => {
      it('handles empty state', async () => {
        const response = await supertest.get(
          `/api/apm/traces?start=${start}&end=${end}&uiFilters=${uiFilters}`
        );

        expect(response.status).to.be(200);
        expect(response.body.items.length).to.be(0);
        expect(response.body.isAggregationAccurate).to.be(true);
      });
    });

    describe('when data is loaded', () => {
      let response: any;
      before(async () => {
        await esArchiver.load(archiveName);
        response = await supertest.get(
          `/api/apm/traces?start=${start}&end=${end}&uiFilters=${uiFilters}`
        );
      });
      after(() => esArchiver.unload(archiveName));

      it('returns the correct status code', async () => {
        expect(response.status).to.be(200);
      });

      it('returns the correct number of buckets', async () => {
        expectSnapshot(response.body.items.length).toMatchInline(`66`);
      });

      it('returns the correct buckets', async () => {
        const sortedItems = sortBy(response.body.items, 'impact');

        const firstItem = sortedItems[0];
        const lastItem = sortedItems[sortedItems.length - 1];

        const groups = sortedItems.map((item) => item.key).slice(0, 5);

        expectSnapshot(sortedItems).toMatch();

        expectSnapshot(firstItem).toMatchInline(`
          Object {
            "averageResponseTime": 3853,
            "impact": 0,
            "key": Object {
              "service.name": "opbeans-ruby",
              "transaction.name": "Api::OrdersController#create",
            },
            "serviceName": "opbeans-ruby",
            "transactionName": "Api::OrdersController#create",
            "transactionType": "request",
            "transactionsPerMinute": 0.016666666666666666,
          }
        `);

        expectSnapshot(lastItem).toMatchInline(`
          Object {
            "averageResponseTime": 1600567.6301369863,
            "impact": 100,
            "key": Object {
              "service.name": "opbeans-python",
              "transaction.name": "opbeans.tasks.sync_customers",
            },
            "serviceName": "opbeans-python",
            "transactionName": "opbeans.tasks.sync_customers",
            "transactionType": "celery",
            "transactionsPerMinute": 1.2166666666666666,
          }
        `);

        expectSnapshot(groups).toMatchInline(`
          Array [
            Object {
              "service.name": "opbeans-ruby",
              "transaction.name": "Api::OrdersController#create",
            },
            Object {
              "service.name": "opbeans-java",
              "transaction.name": "APIRestController#orders",
            },
            Object {
              "service.name": "opbeans-node",
              "transaction.name": "GET /api/types",
            },
            Object {
              "service.name": "opbeans-java",
              "transaction.name": "APIRestController#products",
            },
            Object {
              "service.name": "opbeans-go",
              "transaction.name": "POST /api/orders",
            },
          ]
        `);
      });
    });
  });
}
