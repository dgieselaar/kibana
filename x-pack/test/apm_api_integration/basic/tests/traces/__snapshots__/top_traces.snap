// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top traces when data is loaded returns the correct buckets 1`] = `
Array [
  Object {
    "averageResponseTime": 2577,
    "impact": 0,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /throw-error",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /throw-error",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 3147,
    "impact": 0.06552270160444405,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#orders",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#orders",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 3392.5,
    "impact": 0.09374344413758617,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#order",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#order",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 4713.5,
    "impact": 0.24559517890858723,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#product",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#product",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 4757,
    "impact": 0.25059559560997896,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/products/:id/customers",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/products/:id/customers",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 6787,
    "impact": 0.4839483750082622,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#products",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#products",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 4749.666666666667,
    "impact": 0.5227447114845778,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/orders/:id",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/orders/:id",
    "transactionType": "request",
    "transactionsPerMinute": 0.75,
  },
  Object {
    "averageResponseTime": 7624.5,
    "impact": 0.5802207655235637,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/orders",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/orders",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 5098,
    "impact": 0.582807187955318,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/stats",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/stats",
    "transactionType": "request",
    "transactionsPerMinute": 0.75,
  },
  Object {
    "averageResponseTime": 8181,
    "impact": 0.6441916136689552,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/types/:id",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/types/:id",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 20011,
    "impact": 0.853921734857215,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "POST /api",
    },
    "serviceName": "opbeans-node",
    "transactionName": "POST /api",
    "transactionType": "request",
    "transactionsPerMinute": 0.25,
  },
  Object {
    "averageResponseTime": 6583,
    "impact": 1.2172278724376455,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/products",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/products",
    "transactionType": "request",
    "transactionsPerMinute": 1,
  },
  Object {
    "averageResponseTime": 33097,
    "impact": 1.6060533780113861,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/products/top",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/products/top",
    "transactionType": "request",
    "transactionsPerMinute": 0.25,
  },
  Object {
    "averageResponseTime": 4825,
    "impact": 1.6450221426498186,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#topProducts",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#topProducts",
    "transactionType": "request",
    "transactionsPerMinute": 1.75,
  },
  Object {
    "averageResponseTime": 35846,
    "impact": 1.7640550505645587,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /log-error",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /log-error",
    "transactionType": "request",
    "transactionsPerMinute": 0.25,
  },
  Object {
    "averageResponseTime": 3742.153846153846,
    "impact": 2.4998634943716573,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#customerWhoBought",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#customerWhoBought",
    "transactionType": "request",
    "transactionsPerMinute": 3.25,
  },
  Object {
    "averageResponseTime": 3492.9285714285716,
    "impact": 2.5144049360435208,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET static file",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET static file",
    "transactionType": "request",
    "transactionsPerMinute": 3.5,
  },
  Object {
    "averageResponseTime": 26992.5,
    "impact": 2.8066131947777255,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/types",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/types",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 13516.5,
    "impact": 2.8112687551548836,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/products/:id",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/products/:id",
    "transactionType": "request",
    "transactionsPerMinute": 1,
  },
  Object {
    "averageResponseTime": 20092,
    "impact": 3.168195050736987,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api/customers",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api/customers",
    "transactionType": "request",
    "transactionsPerMinute": 0.75,
  },
  Object {
    "averageResponseTime": 15535,
    "impact": 3.275330415465657,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#stats",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#stats",
    "transactionType": "request",
    "transactionsPerMinute": 1,
  },
  Object {
    "averageResponseTime": 32667.5,
    "impact": 3.458966408120217,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /log-message",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /log-message",
    "transactionType": "request",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 16690.75,
    "impact": 3.541042213287889,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#customers",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#customers",
    "transactionType": "request",
    "transactionsPerMinute": 1,
  },
  Object {
    "averageResponseTime": 33500,
    "impact": 3.5546640380951287,
    "key": Object {
      "service.name": "client",
      "transaction.name": "/customers",
    },
    "serviceName": "client",
    "transactionName": "/customers",
    "transactionType": "route-change",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 77000,
    "impact": 4.129424578484989,
    "key": Object {
      "service.name": "client",
      "transaction.name": "/products",
    },
    "serviceName": "client",
    "transactionName": "/products",
    "transactionType": "page-load",
    "transactionsPerMinute": 0.25,
  },
  Object {
    "averageResponseTime": 19370.6,
    "impact": 5.270496679320978,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "APIRestController#customer",
    },
    "serviceName": "opbeans-java",
    "transactionName": "APIRestController#customer",
    "transactionType": "request",
    "transactionsPerMinute": 1.25,
  },
  Object {
    "averageResponseTime": 81500,
    "impact": 9.072365225837785,
    "key": Object {
      "service.name": "client",
      "transaction.name": "/orders",
    },
    "serviceName": "client",
    "transactionName": "/orders",
    "transactionType": "route-change",
    "transactionsPerMinute": 0.5,
  },
  Object {
    "averageResponseTime": 14419.42857142857,
    "impact": 11.30657439844125,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "ResourceHttpRequestHandler",
    },
    "serviceName": "opbeans-java",
    "transactionName": "ResourceHttpRequestHandler",
    "transactionType": "request",
    "transactionsPerMinute": 3.5,
  },
  Object {
    "averageResponseTime": 270684,
    "impact": 15.261616628971955,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "POST /api/orders",
    },
    "serviceName": "opbeans-node",
    "transactionName": "POST /api/orders",
    "transactionType": "request",
    "transactionsPerMinute": 0.25,
  },
  Object {
    "averageResponseTime": 36010.53846153846,
    "impact": 26.61043592713186,
    "key": Object {
      "service.name": "opbeans-java",
      "transaction.name": "DispatcherServlet#doGet",
    },
    "serviceName": "opbeans-java",
    "transactionName": "DispatcherServlet#doGet",
    "transactionType": "request",
    "transactionsPerMinute": 3.25,
  },
  Object {
    "averageResponseTime": 208000,
    "impact": 35.56882613781033,
    "key": Object {
      "service.name": "client",
      "transaction.name": "/dashboard",
    },
    "serviceName": "client",
    "transactionName": "/dashboard",
    "transactionType": "page-load",
    "transactionsPerMinute": 0.75,
  },
  Object {
    "averageResponseTime": 49816.15625,
    "impact": 91.32732325394932,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "GET /api",
    },
    "serviceName": "opbeans-node",
    "transactionName": "GET /api",
    "transactionType": "request",
    "transactionsPerMinute": 8,
  },
  Object {
    "averageResponseTime": 1745009,
    "impact": 100,
    "key": Object {
      "service.name": "opbeans-node",
      "transaction.name": "Process payment",
    },
    "serviceName": "opbeans-node",
    "transactionName": "Process payment",
    "transactionType": "Worker",
    "transactionsPerMinute": 0.25,
  },
]
`;
